### DISCUSSION TITLE
Swin T 학습

### DISCUSSION THREAD

**Discussion by JiyouSeo** [boostcampaitech2/object-detection-level2-cv-06#109] [Show and tell]
Link: [Swin T 깃링크][gitlink]

[gitlink]: https://github.com/shinya7y/UniverseNet/tree/master/configs/swin_original

<h3 style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; color: rgb(36, 41, 47); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Cascade Mask R-CNN</h3>

Backbone | Pretrain | Lr schd | box AP | mask AP | #params | FLOPs | config | log | model
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
Swin-T | ImageNet-1K | 1x | 48.1 | 41.7 | 86M | 745G | config | github/baidu | github/baidu
Swin-T | ImageNet-1K | 3x | 50.4 | 43.7 | 86M | 745G | config | github/baidu | github/baidu
Swin-S | ImageNet-1K | 3x | 51.9 | 45.0 | 107M | 838G | config | github/baidu | github/baidu
Swin-B | ImageNet-1K | 3x | 51.9 | 45.0 | 145M | 982G | config | github/baidu | github/baidu

현재 Swin-B 3x를 mask-rcnn에 해당하는 부분을 다 걷어내고 학습시키고 있는데, 
config 상에서는 pretrained 모델을 쓰지 않고, lr도  1e-4로 진행하길래
맞춰서 돌려주고 있는데 수렴을 거의 안한다고 봐도 될 것 같네요..36 epoch 일단 돌려놓고 잔 뒤에
내일 결과를 좀 보고 pretrained 모델 사용하고, lr도 좀 올려서 학습 진행해 보아야 할 것 같습니다.(lr을 1e-3으로 진행해봤는데, 발산해서 loss가 NaN이 뜨는 것 같더라구요)
일단 wandb model 내에 cascade_rcnn_swin_base_patch4_window7_mstrain_480-800_giou_4conv1f_adamw_3x_coco_notpre_lr1e-4로 올라가고 있습니다.

**Comment by jaehwan-AI** [comment]
표에 Pretrain = ImageNet-1K가 적혀있는데 pretrain 사용한 것이 아닐까요?
저도 저번에 pretrained model을 사용하지 않았을 때 loss가 발산해서 NaN이 나온 적이 있는데 pretrained weight를 load하니까 해결이 됬었습니다.

**Reply by JiyouSeo** [reply]
네, 표상에는 pretrain이 있는데 config 상에서는 아무리 찾아봐도 pretrained를 사용하지 않더라구요.
pretrained된 모델을 학습하는게 무조건 성능이 높긴 할 것 같지만 혹시 몰라서.. ㅋㅋㅋㅋㅋ(그리고 config에서 mask 부분을 걷어낸게 잘 동작하는지도 볼 수 있구요!)

**Reply by jaehwan-AI** [reply]
저분들은 데이터셋이 충분해서 스크래치로 학습해도 되지만 저희는 데이터셋이 작은 편이라 pretrained를 사용하는 편이 좋을 것으로 생각이 됩니다. pretrained weight를 사용할 수 있는 상황이라면 사용을 추천드립니다.
아마 loss가 한번 발산하면 거의 돌아오지 않는 것으로 알고 있는데 이 부분은 어떻게 됬나요?

**Reply by JiyouSeo** [reply]
lr 1e-3 으로 진행하여 loss가 발산하던 모델은 val부분에서 학습이 멈춰버렸습니다. 
지우님께서 아까 카톡으로 올리신 error가 똑같이 뜨더라구요(ERROR - The testing results of the whole dataset is empty)

**Reply by JiyouSeo** [reply]
네, 오늘 pretrained로 학습 진행하는 코드 추가해서 진행해볼게요 !
